### PR TITLE
send audit result using consistent batch size

### PR DESCRIPTION
**Pull request by alichaddad**

### CODE REVIEW

**Review by khaledkbadr — APPROVED** [review]

**Review by ahsayde — APPROVED** [review]

**Review by amrHassanAbdallah — APPROVED** [review]

**Comment by amrHassanAbdallah** [comment]
[[https://media.giphy.com/media/xSM46ernAUN3y/giphy.gif| ]]

**Comment by amrHassanAbdallah** [comment]
approval gif

File: gateway/governance.go [hunk 1]
```diff
@@ -85,21 +85,32 @@ func (g *MagalixGateway) SetAuditCommandHandler(handler agent.AuditCommandHandle
...
+				g.SendAuditResultsBatch(g.auditResultsBuffer)
+				g.auditResultsBuffer = g.auditResultsBuffer[:0]
+				timer.Reset(auditResultsBatchExpiry)
+			}
+		case <-timer.C:
+			if len(g.auditResultsBuffer) > 0 {
+				g.SendAuditResultsBatch(g.auditResultsBuffer)
+				g.auditResultsBuffer = g.auditResultsBuffer[:0]
+			}
+		}
```
Comment (amrHassanAbdallah) [inline gateway/governance.go:110]: Won't you need to reset the timer here too?
How about having a flush method that does these procedures? send the batch and clear the buff and reset the timer?

Comment (alichaddad) [reply gateway/governance.go:110]: This is a ticker we shouldn't need to reset it, it will run per the interval specified

Comment (amrHassanAbdallah) [reply gateway/governance.go:110]: should we be handling the context whenever it gets closed? so that we send the data before closing the agent maybe?

Comment (alichaddad) [reply gateway/governance.go:110]: Yes that would be better indeed

**Pull request merged**